format/jsontoinflux10.go

```diff
@@ -0,0 +1,175 @@
+
+package format
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/trivago/gollum/core"
+	"github.com/trivago/tgo/tcontainer"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// JSONToInflux10 formatter plugin
+// JSONToInflux10 provides a transformation from JSON data to
+// InfluxDB 0.9.1+ compatible line protocol data.
+// Configuration example
+//
+//  - format.JSONToInflux10
+//      TimeField: time
+//      Measurement: measurement
+//      Ignore:
+//        - ignore
+//        - these
+//        - tags
+//      Tags:
+//        - datacenter
+//        - service
+//        - host
+//        - application
+//
+// TimeField specifies the JSON field that holds the Unix timestamp of the message.
+// The precision is seconds.
+// By default, the field is named `time`. If such a key does not exist in the
+// message, the current Unix timestamp at the time of parsing the message is used.
+//
+// Measurement specifies the JSON field that holds the measurement of the message.
+// By default, the field is named `measurement`. This field MUST exist in the message.
+// If it does not exist in the message, an error will be thrown.
+//
+// Ignore lists all JSON fields that will be ignored and not sent to InfluxDB.
+//
+// Tags lists all names of JSON fields to send to Influxdb as tags.
+// The Influxdb 0.9 convention is that values that do not change every
+// request should be considered metadata and given as tags.
+//
+// JsonToInfluxFormatter defines the formatter applied before the conversion
+// from JSON to InfluxDB. By default this is set to format.Forward.
+type JSONToInflux10 struct {
+	core.SimpleFormatter `gollumdoc:"embed_type"`
+	metadataEscape       *strings.Replacer
+	measurementEscape    *strings.Replacer
+	fieldValueEscape     *strings.Replacer
+	timeField            string `config:"Timefield" default:"time"`
+	measurement          string `config:"Measurement" default:"measurement"`
+	tags                 map[string]bool
+	ignore               map[string]bool
+}
+
+func init() {
+	core.TypeRegistry.Register(JSONToInflux10{})
+}
+
+// Configure initializes this formatter with values from a plugin config.
+func (format *JSONToInflux10) Configure(conf core.PluginConfigReader) {
+	format.SimpleFormatter.Configure(conf)
+
+	format.metadataEscape = strings.NewReplacer(",", "\\,", "=", "\\=", " ", "\\ ")
+	format.measurementEscape = strings.NewReplacer(",", "\\,", " ", "\\ ")
+	format.fieldValueEscape = strings.NewReplacer("\"", "\\\"")
+
+	// Create a set from the tags array for O(1) lookup
+	tagsArray := conf.GetStringArray("Tags", []string{})
+	format.tags = map[string]bool{}
+	for _, v := range tagsArray {
+		format.tags[v] = true
+	}
+
+	// Create a set from the ignore array for O(1) lookup
+	ignoreArray := conf.GetStringArray("Ignore", []string{})
+	format.ignore = map[string]bool{}
+	for _, v := range ignoreArray {
+		format.ignore[v] = true
+	}
+}
+
+func (format *JSONToInflux10) escapeMeasurment(value string) string {
```

Comment:
Why not `value []byte`? If these are meant to be "maybe I want to exchange the implementation in the future" wrappers it would make sense to use the datatype as given by the pipeline?
Otherwise these functions don't make sense to me: Don't solve future problems  + the golang inlining is not that great, yet, i.e. this will create unnecessary overhead.